### PR TITLE
Issue 23 choose from dev, test and prod GCP objects

### LINKED ISSUE #23: add --dev and --test options to the cli
* Add option to run CLI as --dev, --test or --prod (one being the default).

* Go through all instances where config.gcp.dev is invoked, and alter accordingly to use prod, dev or test according to the cli argument

### PR DESCRIPTION
**Pull request by galamit86**
@dkapitan 
Adds functionality to choose between `dev`, `test` and `prod` GCP objects, defined in `config.toml`.

An additional parameter, `gcp_type` is added (next to `id`, `odata_version`, `third_party`, `source` and `config`), and propagated down all relevant functions in `utils.py`. Currently, `gcp_type = dev` is the default. `gcp_type` isn't a perfect name, but I couldn't think of something better.

Current CLI implementation allows to run, from the terminal: `statline-bq --gcp_type=prod` (or `...=test` or `...=dev`). The original idea was `statline-bq --prod`, but when implementing through click, this means defining three boolean flags, and would require an extra check to avoid running with more than one flag set to `True`. Maybe there's another way, not sure if it's worth the trouble.

closes #23

### CODE REVIEW

**Comment by dkapitan** [comment]
@amitgal
Common name for dev, test, prod is environment

**Comment by galamit86** [comment]
@dkapitan 
Nice, how about `gcp_env`? So it's clear it's about gcp? (this all started because I 'm write documentation, I swear ;)

**Comment by dkapitan** [comment]
@galamit86 sounds good to me

File: statline_bq/cli.py [hunk 1]
```diff
@@ -25,15 +32,21 @@ def upload_datasets():
     datasets_path = Path("./datasets.toml")
     config = get_config(config_path)
     datasets = get_datasets(datasets_path)
+    gcp_type = gcp_type.lower()
+    if gcp_type == "dev":
+        gcp_project = config.gcp.dev
```
Comment (dkapitan) [inline statline_bq/cli.py]: @galamit86 
it's good practice to use dicts for case switching in Python, see e.g. [this article](https://medium.com/python-in-plain-english/use-python-dictionary-as-a-switch-2b1f5450d46a)

```
config_envs = {'dev': config.gcp.dev, 'test': config.gcp.test, 'prod': config.gcp.prod}
gcp_project = config_envs.get(gcp_type)
```

**Review by dkapitan — COMMENTED** [review]
see comment about switch case.

File: statline_bq/cli.py (hunk 1 above)
Comment (galamit86) [reply statline_bq/cli.py]: Nice, thanks for the tip!

Implemented, also [here](https://github.com/dataverbinders/statline-bq/blob/a83259c8bd8e71c5bf52620bd5850b103fa56212/statline_bq/utils.py#L378).

**Comment by galamit86** [comment]
@dkapitan 
This PR will remain a draft until #27 is merged (it's branched from it actually). Let me know if you would have a chance to look at #27, or otherwise I'll merge it and move on. I mainly have a feeling that I might be handing over too many parameters (`id`, `odata_version`, `third_party`, `source`, `config` and now also `gcp_env`, but it's more a gut feeling than anything. I am not sure why I don't like it.

**Comment by galamit86** [comment]
@dkapitan 
To avoid a git mess, I'm closing this and reopening - this PR has 17 commits, but 13 of them have been implemented in #27 . I thought it would recognise it, but I am not sure it does, so just in case.

**Pull request closed without merging**